### PR TITLE
Sample and testcase for oauth introspection endpoint

### PR DESCRIPTION
**Pull request by IndunilRathnayake**

### CODE REVIEW

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2ServiceIntrospectionTestCase.java [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.wso2.identity.integration.test.utils.DataExtractUtil.KeyValue;
+
+/**
+ * Created by indunil on 5/27/16.
```
Comment (darshanasbg) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2ServiceIntrospectionTestCase.java]: remove

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2ServiceIntrospectionTestCase.java [hunk 2]
```diff
@@ -0,0 +1,138 @@
+package org.wso2.identity.integration.test.oauth2;
```
Comment (darshanasbg) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2ServiceIntrospectionTestCase.java]: add licences

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2ServiceIntrospectionTestCase.java [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+import java.io.File;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.wso2.identity.integration.test.utils.DataExtractUtil.KeyValue;
+
+public class OAuth2ServiceIntrospectionTestCase extends OAuth2ServiceAbstractIntegrationTest {
```
Comment (omindu) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2ServiceIntrospectionTestCase.java]: Add class level comment.

File: modules/samples/oauth2/playground2/src/main/webapp/oauth2-access-resource.jsp [hunk 4]
```diff
@@ -63,9 +54,10 @@
             	String result = executeGet(resource_url, "", accessTokenIdentifier);
             	session.setAttribute("result", result);
                 response.sendRedirect("user-info.jsp");
-            } else {
-                RequestDispatcher view = request.getRequestDispatcher("my-photos.jsp");
-                view.forward(request, response);
+            } else if (resource_url != null && resource_url.contains("introspect")) {
+                String result = executePost(resource_url, "token=" + accessTokenIdentifier, accessTokenIdentifier);
```
Comment (darshanasbg) [inline modules/samples/oauth2/playground2/src/main/webapp/oauth2-access-resource.jsp]: use different token to authenticate with introspection endpoint

**Comment by omindu** [comment]
Please make improvements to the test case as suggested in https://wso2.org/jira/browse/IDENTITY-4681. Merging since this is urgent.

**Pull request merged**